### DISCUSSION TITLE
24.12.29(일) 백엔드 회의록

### DISCUSSION THREAD

**Discussion by zeus6768** [woowacourse-teams/2024-code-zap#1000] [이야기해봐요]
1. 인프라 작업은 같이 해야 할까요?
   - 운영 서버에 영향 미치는 작업만 같이
2. 검증 로직 책임 분리 (https://github.com/woowacourse-teams/2024-code-zap/pull/988#discussion_r1898477039)
3. 중복된 이름 검증 (https://github.com/woowacourse-teams/2024-code-zap/pull/988#discussion_r1898271871)
    - 에러 메시지 똑같이 가져감 
4. 유니크 제약 조건 (https://github.com/woowacourse-teams/2024-code-zap/pull/988#discussion_r1898293752)
5. var
    - production 환경에서는 var를 전혀 사용하지 않기로 결정
    - test 환경에서는 전부다 var를 사용하기로 결정
    - close #994, #995

**Comment by kyum-q** [comment]
## 삭제 영속성 컨텍스트 안거쳐서 DB 삭제 하는 법

[이전 관련 PR](https://github.com/woowacourse-teams/2024-code-zap/pull/704/commits/b0523112bcc97826b9ac660cb7bd1aed13ede80f)

개선을 위해 필요한 작업
쿼리 최적화
@Modifying을 이용한 삭제 쿼리 개선
템플릿들의 요소를 삭제하는 방법이 현재 deleteByTemplateId 여러번 실행하고 있다.

해당 부분을 한번의 쿼리로 모두 삭제하는 명령어로 변경한다.

@Modifying(clearAutomatically = true)
@Modifying을 사용해 해당 메서드가 데이터베이스의 데이터를 변경하는 쿼리임을 Spring Data JPA에 알릴 수 있다.

영속성 컨텍스트와 데이터 불일치 문제가 생길 수 있다.
영속성 컨텍스트를 비워주는 작업을 도와주는 설정을 추가해야한다.
**@Modifying(clearAutomatically=true) 속성을 사용해 변경 후 자동으로 영속성 컨텍스트를 초기화
벌크 연산 시 사용
벌크 연산은 데이터베이스에서 UPDATE, DELETE 시 대량의 데이터를 한 번에 처리하는 작업
참고 자료 : [@Modifying 이란?](https://hstory0208.tistory.com/entry/JPA-Modifying%EC%9D%B4%EB%9E%80-%EA%B7%B8%EB%A6%AC%EA%B3%A0-%EC%A3%BC%EC%9D%98%ED%95%A0%EC%A0%90-%EB%B2%8C%ED%81%AC-%EC%97%B0%EC%82%B0)
